BUG: Fix crash for 0d timedelta repr

eric-wieser opened the pull request:
Caused by the formatters being changed to take an ND array, rather than a raveled array - tis one was missed.

Also changed to use the builtin isnat, rather than rolling one together where needed.

Fixes #10094

eric-wieser commented on numpy/core/tests/test_arrayprint.py:
```diff
@@ -172,11 +172,23 @@ def test_structure_format(self):
+            texetwrap.dedent("""\
+            [('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',)
+             ('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',) ('NaT',) ('NaT',)
+             ('NaT',) ('NaT',) ('NaT',)]""")
```
@ahaldane: Another formatting change for 1.14 needed here perhaps? Odd how timedelta aligns `NaT`, but datetime does not

ahaldane replied:
It definitely looks like some kind of historical accident, since timedelta format is implemented in python, while the datetime format just falls back to the scalar repr which is coded in C.

I think we should try to make it align if we have time before 1.14.

eric-wieser committed:
TST: Use textwrap.dedent to clarify purpose of test

eric-wieser committed:
BUG: Fix crash in repr of 0d timedelta array

This was missed when the formatters were changed to take an ND.

Also changed to use the builtin isnat, rather than rolling one together where needed.

Fixes #10094

charris commented:
Thanks Eric.
